components/grade-table.js

```diff
@@ -1,23 +1,46 @@
 class GradeTable {
-  constructor(tableElement) {
+  constructor(tableElement, noGradesElement) {
     this.tableElement = tableElement;
+    this.noGradesElement = noGradesElement;
   }
   updateGrades(grades) {
-    console.log(grades);
     var tbody = this.tableElement.querySelector("tbody");
     tbody.textContent = "";
     for (var i = 0; i < grades.length; i++) {
-      var tr = document.createElement("tr");
-      var td1 = document.createElement("td");
-      var td2 = document.createElement("td");
-      var td3 = document.createElement("td");
-      td1.textContent = grades[i].name;
-      td2.textContent = grades[i].course;
-      td3.textContent = grades[i].grade;
-      tr.append(td1);
-      tr.append(td2);
-      tr.append(td3);
-      tbody.append(tr);
+      var renderGrade = this.renderGradeRow(grades[i], this.deleteGrade);
+      tbody.append(renderGrade);
     }
+    if (grades.length > 0) {
+      var pElement = document.getElementById("noGrades");
```

Comment:
Please use the reference saved by the `constructor` instead of re-querying the whole `document`.